src/desktop/dialogs/settingsdialog.cpp

```diff
@@ -124,13 +103,23 @@ SettingsDialog::SettingsDialog(QWidget *parent)
 	m_ui->formLayout_2->removeRow(m_ui->themeChoice);
 #endif
 
-#if defined(HAVE_INPUT_SETTINGS)
-#	if !defined(HAVE_RELATIVE_PEN_MODE_HACK)
-	m_ui->tabletInputCombo->removeItem(INPUT_WINTAB_RELATIVE_PEN_MODE_HACK);
-	m_ui->tabletInputRelativePenModeHackDescription->hide();
+#ifdef Q_OS_WIN
+#	ifdef HAVE_KIS_TABLET
+	m_ui->tabletInputCombo->addItem(
+		tr("KisTablet Windows Ink"), int(tabletinput::Mode::KIS_TABLET_WININK));
+	m_ui->tabletInputCombo->addItem(
+		tr("KisTablet Wintab"), int(tabletinput::Mode::KIS_TABLET_WINTAB));
+	m_ui->tabletInputCombo->addItem(
+		tr("KisTablet Wintab with Relative Pen Mode Hack"),
+		int(tabletinput::Mode::KIS_TABLET_WINTAB_RELATIVE_PEN_HACK));
 #	endif
-#	if defined(INPUT_NO_RESTART_REQUIRED)
-	m_ui->tabletInputRequiresRestartLabel->hide();
+#	if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+	m_ui->tabletInputCombo->addItem(
```

Comment:
One of these Qt6 options must be the equivalent of `Mode::QT5` so I would just have one enumeration value for both of them (like `Default`).

Reply:
Qt5 mode is an undetermined and as far as I know unqueriable superposition of Windows Ink, Wintab and a big wad of nothing, depending on Windows version, Qt version and compile flags. Which doesn't match either Qt6 option enough for me to want to conjoin them.

Reply:
However, the fallback code already makes it equivalent to `Qt6Winink`?

Reply:
It's gotta fall back to *something*, I picked the option that's less broken. Wintab in Qt6 causes menus to only react to pen inputs very sporadically, most of the time they just ignore you.

Conflating Qt6Winink and Qt5MysteryInput into QtDefault would still be possible, but I don't think it would make anything shorter or clearer, all it changes is that that one fallback is implicit. The UI still needs to call them different names and the settings code still has to do different things depending on which Qt version it's under.